File: src/dmdfp/shop/Basket.java
package dmdfp.shop;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by khk on 3/6/14.
 */
public class Basket
{
    private List<BasketItem> items;

    public Basket()
    {
        items = new ArrayList<BasketItem>();
    }

    public List<BasketItem> getItems() {
        return items;
    }

    public void addItem(int itemId, int amount)
    {
        BasketItem item = new BasketItem(itemId, amount);

        items.add(item);
    }

    public void removeItem(int itemId)
    {
        BasketItem item = getItemById(itemId);

        if (item != null)
            items.remove(item);
    }

    public boolean adjustItemAmount(int itemId, int amount)
    {
        BasketItem item = getItemById(itemId);

        if (item != null)
        {
            item.setAmount(amount);
            return true;
        }

        return false;
    }

    public BasketItem getItemById(int itemId)
    {
        for (BasketItem item : items)
        {
            if (item.getItemId() == itemId)
                return item;
        }

        return null;
    }

    public void clear()
    {
        items.clear();
    }
}

File: web/js/basket.js
function Basket()
{
    this.items = [];
}

Basket.prototype.addItem = function(itemId)
{
    for (var i = 0; i < this.items.length; i++) {
        if (itemId == this.items[i].id) {
            this.items[i].amount++;
            this.updateDisplay();
            return;
        }
    }
    this.items.push({ id: itemId, amount: 1 });
    this.updateDisplay();
}

Basket.prototype.removeItem = function(itemId)
{
    for (var i = 0; i < this.items.length; i++)
    {
        if (this.items[i].id == itemId)
        {
            this.items.splice(i, 1);
            return;
        }
    }
}

Basket.prototype.setItems = function(items)
{
    this.items = items;
}

Basket.prototype.getSize = function()
{
    return this.items.length;
}

Basket.prototype.updateDisplay = function()
{
    var list = $("#basket-list");
    list.empty();
    for (var i = 0; i < this.items.length; i++)
    {
        var item = this.items[i];
        list.append($(sprintf('<li>Item id: %d, amount: %d</li>', item.id, item.amount)));
    }
}

Basket.prototype.clearBasket = function() {
    this.items = []
    var list = $("#basket-list");
    list.empty();
    this.updateDisplay();
}
